process/smartContract/vmInput.go

```diff
@@ -100,6 +109,14 @@ func (sc *scProcessor) createVMCallInput(
 	vmCallInput.VMInput.Arguments = finalArguments
 	vmCallInput.GasProvided = vmCallInput.GasProvided - gasLocked
```

Comment:
maybe use here safeSub instead of adding a check with safeAdd afterwards?

Reply:
done. left both of them.